lab_solutions/lab02_08_get_corpus_word_frequencies.py

```diff
@@ -0,0 +1,66 @@
+#! /usr/bin/env python
+# -*- coding: utf-8 -*-
+#
+# Copyright © 2020 Attila Nagy <attila.nagy234@gmail.com>
+#
+# Distributed under terms of the MIT license.
+
+from lab_solutions.lab02_05_word_freq import word_freq
+from lab_solutions.lab02_06_build_wikipedia_corpus import build_wikipedia_corpus
+
+
+def get_and_pretty_print_top_k(freq_dict, k=10):
+    list_to_print = sorted(list(freq_dict.items()), key=lambda x: x[1], reverse=True)[:k]
+    for key, value in list_to_print:
+        print(key + '\t' + str(value))
+
+    return list_to_print
+
+
+def construct_test_case():
+    corpus = []
+    base_text = 'This lecture is really cool. Please give me extra points. I really need them. :)'
+    corpus.append(base_text)
+    animals = ['dog', 'cat', 'bird', 'elephant', 'monkey', 'mice', 'donkey', 'lion', 'medusa', 'whale', 'eagle', 'pig', 'chicken']
+    for i in range(len(animals)):
+        animal_text = ''.join([animals[i] + ' ' for j in range(i+1)])
+        text = f'This sentence contains {i+1} of the same words: ' + str(animal_text)
+        corpus.append(text)
+
+    return corpus
+
+
+def get_corpus_word_frequencies(corpus):
+    base_dict = {}
+    for text in corpus:
+        freqs = word_freq(text)
+        base_dict = {x: base_dict.get(x, 0) + freqs.get(x, 0) for x in set(base_dict.keys()).union(set(freqs.keys()))}
```

Comment:
This is way too difficult to read. Use nested for loops instead:

```suggestion
    for text in corpus:
        for word in text.split(" "):
            # update_base_dict
```